default/python/AI/CombatRatingsAI.py

```diff
@@ -0,0 +1,394 @@
+from collections import Counter
+
+import freeOrionAIInterface as fo  # pylint: disable=import-error
+import sys
+
+import FleetUtilsAI
+from EnumsAI import MissionType
+from freeorion_tools import get_ai_tag_grade, dict_to_tuple, tuple_to_dict, cache_by_session
+from ShipDesignAI import get_part_type
+
+
+def get_empire_standard_fighter():
+    """Get the current empire standard fighter stats, i.e. the most common shiptype within the empire.
+
+    :return: Stats of most common fighter in the empire
+    :rtype: ShipCombatStats
+    """
+    stats_dict = Counter()
+    for fleet_id in FleetUtilsAI.get_empire_fleet_ids_by_role(MissionType.MILITARY):
+        ship_stats = FleetCombatStats(fleet_id, consider_refuel=True).get_ship_combat_stats()
+        stats_dict.update(ship_stats)
+
+    most_commons = stats_dict.most_common(1)
+    if most_commons:
+        return most_commons[0][0]
+    else:
+        return default_ship_stats()
+
+
+def default_ship_stats():
+    """ Return some ship stats to assume if no other intel is available.
+
+    :return: Some weak standard ship
+    :rtype: ShipCombatStats
+    """
+    attacks = (4.0, 1)
+    structure = 15
+    shields = 0
+    fighters = 0
+    launch_rate = 0
+    fighter_damage = 0
+    return ShipCombatStats(stats=(attacks, structure, shields, fighters, launch_rate, fighter_damage))
+
+
+class ShipCombatStats(object):
+    """Stores all relevant stats of a ship for combat strength evaluation."""
+    class BasicStats(object):
+        """Stores non-fighter-related stats."""
+        def __init__(self, attacks, structure, shields):
+            """
+
+            :param attacks:
+            :type attacks: dict|None
```

Comment:
Looks like you expect here a tuple or single value.

`self.attacks = {} if attacks is None else tuple_to_dict(attacks)`